### PR TITLE
Handle fetch node actors correctly in scenario-service

### PR DESCRIPTION
**Pull request by akrmn**
Fixes https://github.com/digital-asset/daml/issues/17729

### CODE REVIEW

File: compiler/damlc/tests/daml-test-files/DivulgeFetchNodeActors.observerAskedToDivulge.EXPECTED.ledger [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  TX 2 1970-01-01T00:00:00Z (DivulgeFetchNodeActors:44:3)
+  #2:0
+  │   disclosed to (since): 'alice' (2), 'divulgee' (2)
+  └─> 'alice' exercises ExeDivulgeFrom on #1:0 (DivulgeFetchNodeActors:DivulgeFrom)
+              with
+                cid = #0:0
+      children:
+      #2:1
+      │   disclosed to (since): 'alice' (2), 'divulgee' (2), 'bank' (2)
+      └─> 'bank' fetches #0:0 (DivulgeFetchNodeActors:Iou)
```
Comment (akrmn) [inline compiler/damlc/tests/daml-test-files/DivulgeFetchNodeActors.observerAskedToDivulge.EXPECTED.ledger]: should be `'alice' fetches #0:0 (DivulgeFetchNodeActors:Iou)`

File: compiler/damlc/tests/daml-test-files/DivulgeFetchNodeActors.observerDivulges.EXPECTED.ledger [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  TX 2 1970-01-01T00:00:00Z (DivulgeFetchNodeActors:44:3)
+  #2:0
+  │   disclosed to (since): 'alice' (2), 'divulgee' (2)
+  └─> 'alice' exercises ExeDivulgeTo on #1:0 (DivulgeFetchNodeActors:DivulgeTo)
+              with
+                cid = #0:0
+      children:
+      #2:1
+      │   disclosed to (since): 'alice' (2), 'divulgee' (2), 'bank' (2)
+      └─> 'bank' fetches #0:0 (DivulgeFetchNodeActors:Iou)
```
Comment (akrmn) [inline compiler/damlc/tests/daml-test-files/DivulgeFetchNodeActors.observerDivulges.EXPECTED.ledger]: should be `'alice' fetches #0:0 (DivulgeFetchNodeActors:Iou)`

**Comment by akrmn** [comment]
I just remembered we can move the `*.EXPECTED.ledger` files to subdirs of `daml-test-files`, commit incoming

**Review by samuel-williams-da — APPROVED** [review]
So the problem was that the ScenarioLedger was putting the actingParties on the nodes, but when we converted it over in the scenario service, we were taking the signatories instead?
If thats the case, happy with the fix :)

**Pull request merged**